mesonbuild/modules/gnome.py

```diff
@@ -188,41 +192,47 @@ def generate_gir(self, state, args, kwargs):
-                elif isinstance(dep.held_object, dependencies.PkgConfigDependency):
-                    for lib in dep.held_object.libs:
-                        if os.path.isabs(lib) and dep.held_object.is_libtool:
-                            scan_command += ["-L%s" % os.path.dirname(lib)]
-                            libname = os.path.basename(lib)
-                            if libname.startswith("lib"):
-                                libname = libname[3:]
-                            libname = libname.split(".so")[0]
-                            lib = "-l%s" % libname
-                        # Hack to avoid passing some compiler options in
-                        if lib.startswith("-W"):
-                            continue
-                            scan_command += [lib]
-
-                    girdir = dep.held_object.get_variable ("girdir")
-                    if girdir:
-                        scan_command += ["--add-include-path=%s" % girdir]
-                else:
-                    mlog.log('dependency %s not handled to build gir files' % dep)
-                    continue
+        deps = (girtarget.get_all_link_deps() + girtarget.get_external_deps() +
+                kwargs.pop('dependencies', []))
+        for dep in deps:
+            if hasattr(dep, 'held_object'):
+                dep = dep.held_object
+            if isinstance(dep, dependencies.InternalDependency):
+                scan_command += self.get_include_args(
+                    state,
+                    dep.include_directories)
+                for lib in dep.libraries:
+                    scan_command += self.get_link_args(state, lib.held_object,
+                                                       depends)
+                for source in dep.sources:
+                    if isinstance(source.held_object, GirTarget):
+                        scan_command += [
+                            "--add-include-path=%s" % (
+                                os.path.join(state.environment.get_build_dir(),
+                                             source.held_object.get_subdir()),
+                            )
+                        ]
+            elif isinstance(dep, dependencies.PkgConfigDependency):
+                scan_command += dep.get_compile_args()
+                for lib in dep.libs:
+                    if os.path.isabs(lib) and dep.is_libtool:
+                        scan_command += ["-L%s" % os.path.dirname(lib)]
+                        libname = os.path.basename(lib)
+                        if libname.startswith("lib"):
+                            libname = libname[3:]
+                        libname = libname.split(".so")[0]
+                        lib = "-l%s" % libname
+                    # Hack to avoid passing some compiler options in
+                    if lib.startswith("-W"):
+                        continue
+                        scan_command += [lib]
```

Comment:
You're right; that's weird, but that's the way it was before. Maybe it's not supposed to be indented this much?